### PR TITLE
feat(vdc): add data-source `cloudavenue_vdc_group` 

### PR DESCRIPTION
**Pull request by gaetanars**
### Description of your changes
add vdc group data-source

If you submit change in the provider code, please make sure to:

- [X] Write or modify examples in `examples/` directory
- [X] Write or modify acceptance tests
- [X] Run `make generate` to ensure the doc was updated properly

### How has this code been tested
```
❯ TF_ACC=1 go test -v -count=1  -run TestAccVDCGroupDataSource ./internal/tests/vdc
=== RUN   TestAccVDCGroupDataSource
--- PASS: TestAccVDCGroupDataSource (4.32s)
PASS
ok      github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/tests/vdc 4.327
```

### CODE REVIEW

File: internal/provider/vdc/vdc_group_datasource.go [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+				Computed:            true,
+			},
+			"network_provider_type": schema.StringAttribute{
+				MarkdownDescription: "Defines the networking provider backing the VDC Group.",
+				Computed:            true,
+			},
+			"status": schema.StringAttribute{
+				MarkdownDescription: "The status that the group can be in 'SAVING', 'SAVED', 'CONFIGURING'," +
+					" 'REALIZED', 'REALIZATION_FAILED', 'DELETING', 'DELETE_FAILED', 'OBJECT_NOT_FOUND'," +
+					" 'UNCONFIGURED'",
```
Comment (azrod) [inline internal/provider/vdc/vdc_group_datasource.go:126]: The list is present 2 times. I suggest to create `const`

```suggestion
				MarkdownDescription: "The status that the group can be in `SAVING`, `SAVED`, `CONFIGURING`," +
					" `REALIZED`, `REALIZATION_FAILED`, `DELETING`, `DELETE_FAILED`, `OBJECT_NOT_FOUND`," +
					" `UNCONFIGURED`",
```

File: internal/provider/vdc/vdc_group_datasource.go [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+				Computed: true,
+			},
+			"type": schema.StringAttribute{
+				MarkdownDescription: "The type of the vDC Group (e.g. 'LOCAL', 'UNIVERSAL').",
```
Comment (azrod) [inline internal/provider/vdc/vdc_group_datasource.go:130]:
```suggestion
				MarkdownDescription: "The type of the vDC Group (e.g. `LOCAL`, `UNIVERSAL`).",
```

File: internal/provider/vdc/vdc_group_datasource.go [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+							Computed:            true,
+						},
+						"is_remote_org": schema.BoolAttribute{
+							MarkdownDescription: "Specifies whether the vDC is local to this VCD site",
+							Computed:            true,
+						},
+						"status": schema.StringAttribute{
+							MarkdownDescription: "The status that the vDC can be in 'SAVING', 'SAVED', 'CONFIGURING'," +
+								" 'REALIZED', 'REALIZATION_FAILED', 'DELETING', 'DELETE_FAILED', 'OBJECT_NOT_FOUND'," +
+								" 'UNCONFIGURED'",
```
Comment (azrod) [inline internal/provider/vdc/vdc_group_datasource.go:157]:
```suggestion
							MarkdownDescription: "The status that the vDC can be in `SAVING`, `SAVED`, `CONFIGURING`," +
					" `REALIZED`, `REALIZATION_FAILED`, `DELETING`, `DELETE_FAILED`, `OBJECT_NOT_FOUND`," +
					" `UNCONFIGURED`",
```

File: internal/tests/vdc/vdc_group_datasource_test.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+}
+`
+
+const dataSourceName = "data.cloudavenue_vdc_group.example"
+
+func TestAccCloudavenueVdcGroupDataSource(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { tests.TestAccPreCheck(t) },
+		ProtoV6ProviderFactories: tests.TestAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
```
Comment (azrod) [inline internal/tests/vdc/vdc_group_datasource_test.go:24]: Put your const in func. 

```suggestion
func TestAccCloudavenueVdcGroupDataSource(t *testing.T) {
	const dataSourceName = "data.cloudavenue_vdc_group.example"
	resource.Test(t, resource.TestCase{
		PreCheck:                 func() { tests.TestAccPreCheck(t) },
		ProtoV6ProviderFactories: tests.TestAccProtoV6ProviderFactories,
		Steps: []resource.TestStep{
```

File: internal/tests/vdc/vdc_group_datasource_test.go [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+			// Read testing
+			{
+				// Apply test
```
Comment (azrod) [inline internal/tests/vdc/vdc_group_datasource_test.go:27]:
```suggestion
```

File: internal/tests/vdc/vdc_group_datasource_test.go [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+				Config: testAccVDCGroupDataSourceConfig,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(dataSourceName, "id", "urn:vcloud:vdcGroup:805a3699-7dba-419b-b731-daae4154617e"),
+				),
```
Comment (azrod) [inline internal/tests/vdc/vdc_group_datasource_test.go:31]: For the computed value please check if attribute is set and for the values passed by user check the value of the attribute

File: templates/data-sources/vdc_group.md.tmpl [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+```terraform
+data "cloudavenue_vdc" "example" {
+	name = "VDC_Frangipane"
+}
+
+data "cloudavenue_vdc_group" "example" {
+  name = data.cloudavenue_vdc.example.vdc_group
+}
+```
```
Comment (azrod) [inline templates/data-sources/vdc_group.md.tmpl:28]:
```suggestion
```terraform
data "cloudavenue_vdc" "example" {
	name = "VDC_Frangipane"
}

data "cloudavenue_vdc_group" "example" {
	name = data.cloudavenue_vdc.example.vdc_group
}
```
```

File: templates/data-sources/vdc_group.md.tmpl [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+{{ .SchemaMarkdown | trimspace }}
+
+{{ if .HasImport -}}
+## Import
+
+Import is supported using the following syntax:
+{{ codefile "shell" .ImportFile }}
```
Comment (azrod) [inline templates/data-sources/vdc_group.md.tmpl:36]: Import is not supported for the DataSources

File: internal/provider/vdc/vdc_group_datasource.go [hunk 9]
```diff
@@ -0,0 +1,275 @@
...
+		resp.Diagnostics.AddError("Error retrieving vDC Group", err.Error())
+		return
+	}
+
+	// populate the data source model with the vdcGroup data
+	data = vdcGroupDataSourceModel{
+		ID:                         types.StringValue(vdcGroup.VdcGroup.Id),
+		Name:                       types.StringValue(vdcGroup.VdcGroup.Name),
+		Description:                types.StringValue(vdcGroup.VdcGroup.Description),
+		DfwEnabled:                 types.BoolValue(vdcGroup.VdcGroup.DfwEnabled),
```
Comment (azrod) [inline internal/provider/vdc/vdc_group_datasource.go:241]: `DFW` is acronym. I suggest to UpperCase `DFW` and add this rule in `.golangci.yml`

File: internal/provider/vdc/vdc_group_datasource.go [hunk 10]
```diff
@@ -0,0 +1,275 @@
...
+		return
+	}
+
+	adminOrg, err := d.client.Vmware.GetAdminOrgByNameOrId(d.client.GetOrg())
+	if err != nil {
+		resp.Diagnostics.AddError("Error retrieving Org", err.Error())
+		return
+	}
+
+	// Get the vdcGroup. If id is not set, user name.
```
Comment (azrod) [inline internal/provider/vdc/vdc_group_datasource.go:220]:
```suggestion
	// Get the vdcGroup. If id is not set, use user name.
```

File: internal/provider/vdc/vdc_group_datasource.go [hunk 11]
```diff
@@ -0,0 +1,275 @@
...
+	if !data.ID.IsNull() && !data.ID.IsUnknown() {
+		vdcGroup, err = adminOrg.GetVdcGroupById(data.ID.ValueString())
+	} else {
+		vdcGroup, err = adminOrg.GetVdcGroupByName(data.Name.ValueString())
+	}
+	if err != nil {
+		if govcd.ContainsNotFound(err) {
+			resp.State.RemoveResource(ctx)
+			return
+		}
```
Comment (azrod) [inline internal/provider/vdc/vdc_group_datasource.go:231]: In the Datasource `RemoveResource` is really necessary ?

File: internal/provider/vdc/vdc_group_datasource.go [hunk 12]
```diff
@@ -0,0 +1,275 @@
...
+
+		return
+	}
+
+	d.client = client
+}
+
+func (d *vdcGroupDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var data vdcGroupDataSourceModel
+	var diag diag.Diagnostics
```
Comment (azrod) [inline internal/provider/vdc/vdc_group_datasource.go:205]:
```suggestion
	var (
		data vdcGroupDataSourceModel
		diag diag.Diagnostics
	)
```

File: internal/provider/vdc/vdc_group_datasource.go [hunk 13]
```diff
@@ -0,0 +1,275 @@
...
+				},
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: "The name of the vDC Group.",
+				Optional:            true,
+				Computed:            true,
+				Validators: []validator.String{
+					stringvalidator.ExactlyOneOf(path.MatchRoot("id")),
+				},
+			},
```
Comment (azrod) [inline internal/provider/vdc/vdc_group_datasource.go:94]: In the terraform documentation generator the `ID` attribute is only `Computed`. 
Suddenly in the documentation the attribute is in `Read-Only` section. 

I suggest to create new attribute `vdc_group_id`

```suggestion
			"id": schema.StringAttribute{
				// Add MarkdownDescription
				Required: true,
			},
			"vdc_group_id": schema.StringAttribute{
				Optional: true,
				Computed: true,
				Validators: []validator.String{
					stringvalidator.ExactlyOneOf(path.MatchRoot("vdc_group_name")),
				},
			},
			"vdc_group_name": schema.StringAttribute{
				MarkdownDescription: "The name of the vDC Group.",
				Optional:                true,
				Computed:            true,
				Validators: []validator.String{
					stringvalidator.ExactlyOneOf(path.MatchRoot("vdc_group_id")),
				},
			},
```

**Review by azrod — CHANGES_REQUESTED** [review]

**Comment by gaetanars** [comment]
```
❯ TF_ACC=1 go test -v -count=1  -run TestAccVDCGroupDataSource ./internal/tests/vdc
=== RUN   TestAccVDCGroupDataSource
--- PASS: TestAccVDCGroupDataSource (4.32s)
PASS
ok      github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/tests/vdc 4.327s
```

**Review by azrod — APPROVED** [review]

**Pull request merged**